cli/initiator/initiator.go

```diff
@@ -106,46 +114,71 @@ var StartDKG = &cobra.Command{
 		logLevelFormat := viper.GetString("logLevelFormat")
 		logFilePath := viper.GetString("logFilePath")
 		// If the log file doesn't exist, create it
-		_, err = os.OpenFile(logFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
-		if err != nil {
-			return err
+		if _, err := os.Stat(logFilePath); os.IsNotExist(err) {
+			_, err := os.Create(logFilePath)
+			if err != nil {
+				return err
+			}
 		}
 		if err := logging.SetGlobalLogger(logLevel, logFormat, logLevelFormat, &logging.LogFileOptions{FileName: logFilePath}); err != nil {
 			return fmt.Errorf("logging.SetGlobalLogger: %w", err)
 		}
 		logger := zap.L().Named("dkg-initiator")
 		// Check paths for results
-		depositOutputPath := viper.GetString("depositOutputPath")
-		if depositOutputPath == "" {
+		outputPath := viper.GetString("outputPath")
+		if outputPath == "" {
 			logger.Fatal("😥 Failed to get deposit result path flag value: ", zap.Error(err))
 		}
-		_, err = os.Stat(depositOutputPath)
-		if os.IsNotExist(err) {
-			logger.Fatal("😥 Folder to store deposit file does not exist: ", zap.Error(err))
-		}
-		// Check paths for results
-		keysharesOutputPath := viper.GetString("keysharesOutputPath")
-		if keysharesOutputPath == "" {
-			logger.Fatal("😥 Failed to get ssv payload path flag value: ", zap.Error(err))
-		}
-		_, err = os.Stat(keysharesOutputPath)
-		if os.IsNotExist(err) {
-			logger.Fatal("😥 Folder to store SSV payload file does not exist: ", zap.Error(err))
+		if stat, err := os.Stat(outputPath); err != nil || !stat.IsDir() {
+			logger.Fatal("😥 Error to to open path to store results", zap.Error(err))
 		}
 		// Load operators TODO: add more sources.
-		operatorFile := viper.GetString("operatorsInfoPath")
-		if operatorFile == "" {
-			logger.Fatal("😥 Failed to get operator info file path flag value: ", zap.Error(err))
+		operatorsInfo := viper.GetString("operatorsInfo")
+		operatorsInfoPath := viper.GetString("operatorsInfoPath")
+		if operatorsInfo == "" && operatorsInfoPath == "" {
+			logger.Fatal("😥 Operators string or path have not provided")
 		}
-
-		opsfile, err := os.ReadFile(operatorFile)
-		if err != nil {
-			logger.Fatal("😥 Failed to read operator info file: ", zap.Error(err))
+		if operatorsInfo != "" && operatorsInfoPath != "" {
+			logger.Fatal("😥 Please provide either operator info string or path, not both")
 		}
-
-		opMap, err := initiator.LoadOperatorsJson(opsfile)
-		if err != nil {
-			logger.Fatal("😥 Failed to load operators: ", zap.Error(err))
+		var opMap initiator.Operators
+		if operatorsInfo != "" {
+			logger.Info("📖 reading raw JSON string of operators info", zap.String("", operatorsInfo))
+			opMap, err = initiator.LoadOperatorsJson([]byte(operatorsInfo))
+			if err != nil {
+				logger.Fatal("😥 Failed to load operators: ", zap.Error(err))
+			}
+		}
+		if operatorsInfoPath != "" {
```

Comment:
what if both of them are empty?

Reply:
We have a check for it: https://github.com/bloxapp/ssv-dkg/blob/after_demo_fixes/cli/initiator/initiator.go#L138